### DISCUSSION TITLE
Flux

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [BottlecapDave/HomeAssistant-OctopusEnergy#237] [General]
Any plans to make this work for Flux accounts?

**Comment by BottlecapDave** [comment]
If flux tariffs follow usual structures, then they should already be supported. A few users have confirmed they use this integration with flux (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/209). Can you elaborate on what is/isn't working with flux accounts?

**Reply by Captain0Kangaroo** [reply]
Ah right...I've not actually tried it yet but i wasn't sure if it would setup the 3 tariffs or not as it didn't seem to elude that it would.

**Reply by BottlecapDave** [reply]
It pulls down the rates that OE provide and expose this via rate sensors. If you setup target rate sensors, then these will target the best rates (either import or export depending on what meter you're targeting). The part that it might not support is the new off peak/peak sensors as this is only applicable when your tariff provides two rates during a single 24 hour period.

**Reply by BottlecapDave** [reply]
But from a basic level, the integration has been built to support all tariffs that OE provide as long as they're all exposed in the same way via their API. However, some tariffs might provide additional functionality that isn't supported without additional work (e.g. intelligent).

**Reply by furstyferret-dev** [reply]
For info - I have Flux and this integration works perfectly.

**Reply by Captain0Kangaroo** [reply]
> For info - I have Flux and this integration works perfectly. 

How are you using for Flux?

Ideally you'd charge your battery overnight so you start exporting as early as poss. 

The problem I have is I have to set force charge every night to make it work for Flux and this is not always something I remember so recently I disabled it which is a shame as I like all the stats it give. 

I'm I missing something?

**Comment by furstyferret-dev** [comment]
I have east/west panels so they fill the battery quickly in the morning
and generate a fair amount during the peak export period. I don’t force the
batteries to discharge during this time.

From an EV point of view I just have it charge during the 3 hour cheap
period and accept a small loss on the export if I top up from solar during
the day. I get the moral high ground, if not the financial one.

Overall though it’s worth it compared to the SEG in the summer.

Are you not able to automatically schedule the charging?

**Reply by lesma** [reply]
Can I ask if the cost is working for you with a flux tariff? For me it seems to be adding peak and flux together than then calculating the peak cost based on this combined flux and peak usage.

**Reply by RobinXe** [reply]
Your standpoint may have already changed, this far into the winter, but have you done a comparison with Intelligent? On my initial back-of-a-fag-packet calculation, you'd have to export an awful lot on Flux to make it preferable to charging the battery overnight on IO, and exporting any excess at twice the import rate (now that IO permits export on Fixed @ 15 p/kWh).

**Reply by BottlecapDave** [reply]
Flux calculations are still applicable as described in https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/237#discussioncomment-5875077 and the same as [Cosy](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/608) and is by design for the moment. However, with more tariffs coming out that are now having off peak, peak and super peak, this will probably need some rethink in the future (but is not currently on the agenda)

**Reply by lesma** [reply]
Fair enough thanks for the update, just wanted to make sure I was not missing something with my setup.

For me the cheaper 3 hours off peak does make flux cheaper overall even this time of year with very little solar. I do have a 11.5kWh battery though, so do imagine it will vary based on general energy use and battery size.
